packages/nuxt/src/components/islandsTransform.ts

```diff
@@ -1,86 +1,122 @@
 import { pathToFileURL } from 'node:url'
+import { join } from 'node:path'
+import fs from 'node:fs'
 import type { Component } from '@nuxt/schema'
 import { parseURL } from 'ufo'
 import { createUnplugin } from 'unplugin'
 import MagicString from 'magic-string'
 import { ELEMENT_NODE, parse, walk } from 'ultrahtml'
+import { hash } from 'ohash'
+import { resolvePath } from '@nuxt/kit'
 import { isVue } from '../core/utils'
 
 interface ServerOnlyComponentTransformPluginOptions {
     getComponents: () => Component[]
+    /**
+     * passed down to `NuxtTeleportSsrClient`
+     * should be done only in dev mode as we use build:manifest result in production
+     */
+    rootDir?: string
+    isDev?: boolean
+}
+
+interface ComponentChunkOptions {
+  getComponents: () => Component[]
+  buildDir: string
 }
 
 const SCRIPT_RE = /<script[^>]*>/g
-const HAS_SLOT_RE = /<slot[^>]*>/
+const HAS_SLOT_OR_CLIENT_RE = /(<slot[^>]*>)|(nuxt-client)/
 const TEMPLATE_RE = /<template>([\s\S]*)<\/template>/
+const NUXTCLIENT_ATTR_RE = /\snuxt-client(="[^"]*")?/g
 
-export const islandsTransform = createUnplugin((options: ServerOnlyComponentTransformPluginOptions) => {
+export const islandsTransform = createUnplugin((options: ServerOnlyComponentTransformPluginOptions, meta) => {
+  const components = options.getComponents()
+  const isVite = meta.framework === 'vite'
+  const { isDev, rootDir } = options
   return {
     name: 'server-only-component-transform',
     enforce: 'pre',
     transformInclude (id) {
       if (!isVue(id)) { return false }
 
-      const components = options.getComponents()
```

Comment:
IIRC I think this is needed here if the the components _change_ after the start of Nuxt.